File: python/difference-of-squares/difference_of_squares.py
def difference_of_squares(n: int) -> int:
    'Return (1 + 2 + ... + n)² - (1² + 2² + ... + n²)'
    return square_of_sum(n) - sum_of_squares(n)

def square_of_sum(n: int) -> int:
    'Return (1 + 2 + ... + n)²'
    return (n*(n + 1) // 2) ** 2

def sum_of_squares(n: int) -> int:
    'Return (1² + 2² + ... + n²)'
    return sum(x*x for x in range(1, n+1))

File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  module Squarable
    refine Integer do
      def square
        self * self
      end
    end
  end
  using Squarable

  def initialize(number)
    @number = number
  end

  attr_reader :number

  def difference
    square_of_sum - sum_of_squares
  end

  def square_of_sum
    (number * (number + 1) / 2).square
  end

  def sum_of_squares
    (1..number).map(&:square).sum
  end
end
